.gitignore

```diff
@@ -0,0 +1 @@
+.idea
```

Comment:
Please add a comment to explain what this is (see format of https://github.com/kubernetes/kubernetes/blob/master/.gitignore).

Reply:
Copied over all IDE related ignore

Reply:
Thanks for the review.
Copied over all IDE related ignore config from the kubernetes repo and OSX files.